src/test/java/org/opentripplanner/middleware/cdp/ConnectedDataPlatformTest.java

```diff
@@ -0,0 +1,241 @@
+            tripRequest = null;
+        }
+        if (tripSummary != null) {
+            Persistence.tripSummaries.removeById(tripSummary.id);
+            tripSummary = null;
+        }
+        for(TripHistoryUpload tripHistoryUpload : Persistence.tripHistoryUploads.getAll()) {
+            Persistence.tripHistoryUploads.removeById(tripHistoryUpload.id);
+        }
+        if (tempFile != null) {
+            FileUtils.deleteFile(new File(tempFile));
+            tempFile = null;
+        }
+        if (zipFileName != null) {
+            S3Utils.deleteObject(
+                ConnectedDataManager.CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME,
+                String.format("%s/%s", ConnectedDataManager.CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME, zipFileName)
+            );
+            zipFileName = null;
+        }
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        setAuthDisabled(false);
+        if (otpUser != null) {
+            Persistence.otpUsers.removeById(otpUser.id);
+        }
+    }
+
+    /**
+     * Make sure that the first upload is created and contains the correct upload date.
+     */
+    @Test
+    public void canStageFirstUpload() {
+        ConnectedDataManager.stageUploadDays();
+        TripHistoryUpload tripHistoryUpload = TripHistoryUpload.getFirst();
+        Date startOfDay = getStartOfDay(getDateMinusNumberOfDays(new Date(), 1));
+        assertNotNull(tripHistoryUpload);
+        assertEquals(startOfDay.getTime(), tripHistoryUpload.uploadDate.getTime());
+    }
+
+    /**
+     * Confirm that a single zip file is created which contains a single JSON file. Also confirm that the contents
+     * written to the JSON file is correct and covers a single day's worth of trip data.
+     */
+    @Test
+    public void canCreateZipFileWithContent() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        ConnectedDataManager.IS_TEST = true;
+
+        String userId = UUID.randomUUID().toString();
+        Date startOfYesterday = getStartOfDay(getDateMinusNumberOfDays(new Date(), 1));
+        tripRequest = PersistenceTestUtils.createTripRequest(userId, startOfYesterday);
+        tripSummary = PersistenceTestUtils.createTripSummary(tripRequest.id, startOfYesterday);
+        ConnectedDataManager.stageUploadDays();
+        ConnectedDataManager.processTripHistory();
+        zipFileName = getFileName(startOfYesterday, ConnectedDataManager.ZIP_FILE_NAME_SUFFIX);
+        tempFile = String.join(
+            "/",
+            FileUtils.getTempDirectory().getAbsolutePath(),
+            zipFileName
+        );
+        String fileContents = getContentsOfFileInZip(
+            tempFile,
+            getFileName(startOfYesterday, ConnectedDataManager.DATA_FILE_NAME_SUFFIX)
+        );
+        String contents = ConnectedDataManager.getAnonymizedTripHistory(startOfYesterday, getEndOfDay(startOfYesterday));
+        assertEquals(fileContents, contents);
```

Comment:
This doesn't seem to be a great assertion here since a subcomponent of the `processTripHistory` method is being called to generate the expected data. So long as the implementation remains the same, it's always going to match and thus what you're really testing is anything up until the calling of `getAnonymizedTripHistory`. In this case, I'd recommend creating a snapshot of the data to assert against. See example of snapshotting in gtfs-lib: https://github.com/conveyal/gtfs-lib/blob/94ff5fda5785020e3a689e8f5daf33291b4b7625/src/test/java/com/conveyal/gtfs/graphql/GTFSGraphQLTest.java#L99